src/common/arrow/arrow_row_batch.cpp

```diff
@@ -106,82 +164,179 @@ void ArrowRowBatch::templateCopyNonNullValue<bool>(
 }
 
 template<>
-void ArrowRowBatch::templateCopyNonNullValue<std::string>(
-    ArrowVector* vector, Value* value, std::int64_t pos) {
+void ArrowRowBatch::templateCopyNonNullValue<STRING>(
+    ArrowVector* vector, const main::DataTypeInfo& typeInfo, Value* value, std::int64_t pos) {
     auto offsets = (std::uint32_t*)vector->data.data();
     auto strLength = value->strVal.length();
     offsets[pos + 1] = offsets[pos] + strLength;
     vector->overflow.resize(offsets[pos + 1]);
     std::memcpy(vector->overflow.data() + offsets[pos], value->strVal.data(), strLength);
 }
 
-void ArrowRowBatch::copyNonNullValue(ArrowVector* vector, Value* value, std::int64_t pos) {
-    switch (value->dataType.typeID) {
+template<>
+void ArrowRowBatch::templateCopyNonNullValue<LIST>(
+    ArrowVector* vector, const main::DataTypeInfo& typeInfo, Value* value, std::int64_t pos) {
+    vector->data.resize((pos + 2) * sizeof(std::uint32_t));
+    auto offsets = (std::uint32_t*)vector->data.data();
+    auto numElements = value->listVal.size();
+    offsets[pos + 1] = offsets[pos] + numElements;
+    auto numChildElements = offsets[pos + 1] + 1;
+    auto currentNumBytesForChildValidity = vector->childData[0]->validity.size();
+    auto numBytesForChildValidity = getNumBytesForBits(numChildElements);
```

Comment:
Maybe add a comment here about what is `ChildValidity` and the meaning of `0xFF`.